### DISCUSSION TITLE
Using interpolating functions in constraints with derivatives

### DISCUSSION THREAD

**Discussion by notesofdabbler** [infiniteopt/InfiniteOpt.jl#212] [Q&A] [answered]
I am trying a toy ODE problem of simulating volume in a tank when flowrate into the tank follows a set time profile. I was putting a constraint of the following form which gave the error below (qint(t) is an interpolation function)
<img width="1028" alt="Screen Shot 2022-01-30 at 10 40 33 PM" src="https://user-images.githubusercontent.com/7705554/151735727-0f7563a4-ead5-4220-93d9-67d41463624b.png">
The notebook which has this example is in this [location](https://github.com/notesofdabbler/learn_Julia/blob/master/julia_paresto/test_interpolation_fn.ipynb). I wanted to check if using interpolation functions is not currently supported.

**Comment by pulsipher** [comment, marked as answer]
For functions that depend only on infinite parameters such as this one, we can use the parameter function syntax (https://pulsipher.github.io/InfiniteOpt.jl/stable/guide/expression/#par_func_docs). In this case you can do:
```julia
@parameter_function(m, qint_func == t -> qint(t))
@constraint(m, deriv(V, t) == qint_func)
```

**Comment by notesofdabbler** [comment]
Thanks, that fixed it.
